ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -306,6 +311,44 @@ class ContractsService(
     }
   }
 
+  def refreshCache(
+      jwt: Jwt,
+      jwtPayload: JwtPayload,
+      refreshCacheRequest: RefreshCacheRequest,
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      metrics: HttpJsonApiMetrics,
+  ): Future[SearchResult[Error \/ domain.RefreshCacheResult]] = {
```

Comment:
It's not a `SearchResult`, so we shouldn't re-use that type. For error reporting, it might be worth adding and extra `errors` field to `RefreshCacheResult`.

```suggestion
  ): Future[domain.RefreshCacheResult[Error \/ domain.Offset] = {
```

Reply:
Yeah, but that type is customised for search results, which is not appropriate for us here. I'm saying we should use our own type.

Reply:
we always need to return a SyncResponse.... the SearchResult is only an alias for it....
Why we always use SyncResponse?? because by design the API is not RESTful so the return body requires an status code(which is embebed in the SyncResponse)